### PR TITLE
Button

### PR DESCRIPTION
**Pull request by Arbina001**
### List Button

Screenshot 2023-12-19 at 4 33 56 PM

### Outline Button
Screenshot 2023-12-19 at 4 34 21 PM

### Icon button
Screenshot 2023-12-19 at 4 34 40 PM

### Text Button 
Screenshot 2023-12-19 at 4 34 57 PM

### CODE REVIEW

File: src/stories/button/Buttons.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  return (
+    <StyledButtonComponent
+      type={type}
+      variant={variant}
+      isOutline={isOutline}
+      border={border}
+      disableHover={disableHover}
+    >
+      {icon} {text}
+    </StyledButtonComponent>
```
Comment (shubham-spreadd) [inline src/stories/button/Buttons.js:23]: outline border & hover should be managed by variant.

File: src/stories/button/Buttons.stories.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+export default {
+  title: "Buttons",
+  component: Button,
+};
+
+
+export const listButton = () => (
+  <StyledContainer>
+    <Button type="primary" variant="contained" text="PRIMARY BUTTON" disableHover />
+    <Button type=" iconAfterLinkColor" variant="contained" text="SECONDARY BUTTON" disableHover  />
```
Comment (shubham-spreadd) [inline src/stories/button/Buttons.stories.js:17]: can you please explain this type `iconAfterLinkColor`?

File: webpack.config.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ module.exports={
           directory: path.join(__dirname, 'dist'),
         },
         compress: true,
-        port: 5000,
+        port: 1234,
```
Comment (shubham-spreadd) [inline webpack.config.js:16]: any reason for this change?

**Review by shubham-spreadd — CHANGES_REQUESTED** [review]

File: src/stories/button/Buttons.stories.js (hunk 2 above)
Comment (Arbina001) [reply src/stories/button/Buttons.stories.js:17]: Sir it's the color name of button which i added in theme file

File: src/stories/button/Buttons.style.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+
+  font-size: 1rem;
+  font-weight: 600;
+  background-color: ${({ choosecolor }) =>
+    choosecolor === "secondary" ? theme.basiccolor : theme?.[choosecolor]};
+
+
+  color: ${({ choosecolor }) =>
+    choosecolor === "secondary" ? theme?.[choosecolor] : theme.basiccolor};
```
Comment (shubham-spreadd) [inline src/stories/button/Buttons.style.js:19]: The variable name "choosecolor" isn't clear. Could you consider renaming it to something more descriptive and relevant?

**Review by shubham-spreadd — CHANGES_REQUESTED** [review]

**Review by shubham-spreadd — APPROVED** [review]

File: src/stories/button/Button.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import {
+  StyledButtonComponent,
+  OutlineButtons,
+  TextButtons 
+} from './Button.style';
+
+const Button = ({ type, variant, text, icon }) => {
```
Comment (tarakhant1297) [inline src/stories/button/Button.js:8]: @Arbina001 use color props instead of type

File: src/stories/button/Button.style.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+export const StyledButtonComponent = styled.button`
+  text-transform: uppercase;
+  border: 2px solid ${theme.white};
+  font-style: italic;
+
+  font-size: 1rem;
+  font-weight: 600;
+  background-color: ${({ type }) =>
+    type === 'secondary' ? theme.basiccolor : theme?.[type]};
```
Comment (tarakhant1297) [inline src/stories/button/Button.style.js:12]: @Arbina001 can you please explain this code

File: src/theme.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+//
+// theme.js
```
Comment (tarakhant1297) [inline src/theme.js:2]: @Arbina001 remove commented code

**Review by tarakhant1297 — CHANGES_REQUESTED** [review]

File: src/stories/button/Button.style.js (hunk 6 above)
Comment (Arbina001) [reply src/stories/button/Button.style.js]: this is white color which i have used in my theme

File: src/stories/button/Button.style.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+  font-size: 1rem;
+  font-weight: 600;
+  background-color: ${({ color }) =>
+    color === 'secondary' ? theme.whitecolor : theme?.[color]};
+
+  color: ${({ color }) =>
+    color === 'secondary' ? theme?.[color] : theme.whitecolor};
+
+  border-top-left-radius: 2px;
```
Comment (ankitadrixus08) [inline src/stories/button/Button.style.js:17]: give shorthand property here @Arbina001

File: src/stories/button/Button.style.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
+import styled, { css } from 'styled-components';
+import theme from '../../theme';
+
+export const StyledButtonComponent = styled.button`
+  text-transform: uppercase;
+  border: 2px solid ${theme.white};
+  font-style: italic;
+
+  font-size: 1rem;
```
Comment (ankitadrixus08) [inline src/stories/button/Button.style.js:9]: 1 rem = how much px? @Arbina001

File: src/theme.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+const theme = {
+  primary: "#FF4E54",
+  secondary: "#4a4a68",
+  success: "#07CB6E",
+  warning: "#F3B664",
+  cardBorderColor: "#E49393",
+  black: "#060047",
+  cardTextColor: "#1640D6",
```
Comment (ankitadrixus08) [inline src/theme.js:8]: color name should not give like as per component it should be as per color name like dark-gray, yellowGreen, lightBlack @Arbina001

File: src/stories/button/Button.style.js (hunk 9 above)
Comment (Arbina001) [reply src/stories/button/Button.style.js:9]: 16 px

File: src/stories/button/Button.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+      <OutlineButtons color={color}>
+        {icon} {text}
+      </OutlineButtons>
+    );
+  }
+  if (type === 'text') {
+    return <TextButtons color={color}>{text}</TextButtons>;
+  }
+
+  return null;
```
Comment (tarakhant1297) [inline src/stories/button/Button.js:28]: @Arbina001  you return direct this instead  of null and remove if condition`return <TextButtons color={color}>{text}</TextButtons>;`

**Review by tarakhant1297 — CHANGES_REQUESTED** [review]

File: src/theme.js (hunk 10 above)
Comment (Arbina001) [reply src/theme.js:8]: ok sir I have change it in main file

**Pull request merged**